### DISCUSSION TITLE
rowActions query override

### DISCUSSION THREAD

**Discussion by EldoonNemar** [Okipa/laravel-table#154] [Q&A]
Is it possible to override the rowActions model so that it doesn't run the default closure query by id? I'm leveraging softDelete, and even though I'm calling onlyTrashed() within the table query, the action query doesn't override the model so it always returns a 404 because the query is only searching non-trashed results on the model.

**Comment by EldoonNemar** [comment]
Example:
```
       return Table::make()->model(FileUpload::class)
        ->query(function($query) {
            $query->onlyTrashed();
        })
        ->rowActions(fn(FileUpload $data) => [
                new RestoreFileAction(),
        ];
```

RestoreFileAction is nearly identical to DestroyRowAction but with ->restore() in the action function. The issue is, the FileUpload closure in rowActions uses the default  query of 

> select * from "file_uploads" where "file_uploads"."id" = ? and "file_uploads"."deleted_on" is null limit 1 ["1260097"]

the query onlyTrashed() should (and does for the table display only) make the query file_uploads.deleted_on is not null

The initial count query works a treat with onlyTrashed;
> select count(*) as aggregate from "file_uploads" left join "scepter"."search" on "search"."file_detail_id" = "file_uploads"."file_detail_id" left join "hash_types" on "hash_types"."id" = "search"."file_type" where "file_uploads"."deleted_on" is not null and "file_uploads"."uploaded_by" = ? and "file_uploads"."deleted_on" is not null [176]

TLDR; i need the model closure for rowActions to include onlyTrashed() also
